camera/camera_manager.cc

```diff
@@ -16,68 +50,41 @@ CameraManager::CameraManager() {
 }
 
 void CameraManager::load_configs() {
-  // directory iteration from https://stackoverflow.com/questions/612097/how-can-i-get-the-list-of-files-in-a-directory-using-c-or-c/37494654#37494654
+  // directory iteration from
+  // https://stackoverflow.com/questions/612097/how-can-i-get-the-list-of-files-in-a-directory-using-c-or-c/37494654#37494654
   const std::string config_dir = "/jet/camera/cfg";
-  for (const auto & entry : fs::directory_iterator(config_dir)) {
+  for (const auto& entry : fs::directory_iterator(config_dir)) {
     const std::string filename = entry.path();
-    if (filename == "." || filename == ".."){ 
+    if (filename == "." || filename == "..") {
       continue;
     }
-    std::optional<YAML::Node> config = read_YAML(filename);
-    std::optional<Camera> camera = parse_config(*config);
+
+    const std::optional<YAML::Node> config = read_YAML(filename);
+    if (!config) {
+      throw std::runtime_error("Could not open config");
+    }
+
+    const std::optional<Camera> camera = extract_v4l_path(*config);
     if (camera.has_value()) {
       camera_map_[camera->serial_number] = *camera;
     }
-  }
-}
 
-std::optional<YAML::Node> CameraManager::read_YAML(const std::string& filepath) const {
-  try {
-    return YAML::LoadFile(filepath);
-  } catch (YAML::BadFile e) {
-    const std::string err = std::string("Could not find YAML file ") + filepath;
-    throw std::runtime_error(err);
+    const auto calibration = extract_calibration(*config);
+    calibration_map_[camera->serial_number] = calibration;
   }
 }
 
-std::optional<Camera> CameraManager::parse_config(const YAML::Node& cfg) const {
-  Camera camera;
-  // path & video index
-  camera.serial_number = cfg["serial_number"].as<std::string>();
-  camera.v4l_path = "/dev/v4l/by-id/" + cfg["v4l_path"].as<std::string>();
-  const auto optional_camera_index = follow_v4l_path(camera.v4l_path);
-  
-  if (!optional_camera_index.has_value()){
-    return std::nullopt;
-  }
-  else{
-    camera.video_index = optional_camera_index.value();
+Calibration CameraManager::get_calibration(const std::string& serial_number) const {
+  if (calibration_map_.count(serial_number) == 0) {
+    const std::string err = std::string("Camera serial number is not known.");
```

Comment:
Maybe specify where you looked for and did not find the serial number, eg "no config contains this serial number"